network/framework/src/application/interface.rs

```diff
@@ -196,8 +200,13 @@ impl<Message: NetworkMessageTrait> NetworkClientInterface<Message> for NetworkCl
         unimplemented!("Adding peers to discovery is not yet supported!");
     }
 
-    async fn disconnect_from_peer(&self, peer: PeerNetworkId) -> Result<(), Error> {
+    async fn disconnect_from_peer(
+        &self,
+        peer: PeerNetworkId,
+        reason: DisconnectReason,
+    ) -> Result<(), Error> {
         let network_sender = self.get_sender_for_network_id(&peer.network_id())?;
+        warn!("Disconnecting from peer with id {}: {}", peer, reason);
```

Comment:
We should also verify that an appropriate metrics event is being emitted on disconnects (with the reason as a label). Would be good to verify disconnect behaviour visually (i.e., without having to scan through logs). 

We can leave this as a follow up 😄